### PR TITLE
Add new glanceAPI struct

### PR DESCRIPTION
**Pull request by fmount**
This change is the first of a series where the purpose is to move away from the harcoded internal/external `GlanceAPI` instance in favor of an arbitrary number of `GlanceAPI` deployments.
The current change is based on [1] that does not introduce the ability to deploy >1 `GlanceAPI` instances, but introduces the a single `glanceAPI` struct that will serve this purpose.
The split behavior is still preserved, but it happens behind the scenes in Glance.
Finally, this patch bumps the operators to make sure everything works with the latest version of the API.

[1]  https://github.com/openstack-k8s-operators/glance-operator/pull/329

### CODE REVIEW

**Comment by viroel** [comment]
recheck

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
@viroel I think I just had a wrong `Depends-On` in the PR description: it actually built the `glance-operator` as dependency for the meta-operator and I see it here [1].
Also, I can see the crc deployment adopting the new `glanceAPI` CRD and build the ctlplane accordingly.

[1] https://logserver.rdoproject.org/14/514/623c39ba67a1e28621ca036c6745c202cfb17148/github-check/openstack-k8s-operators-content-provider/8a6dc42/ci-framework-data/logs/ 
[2] https://logserver.rdoproject.org/14/514/623c39ba67a1e28621ca036c6745c202cfb17148/github-check/podified-multinode-edpm-deployment-crc/da24d2b/controller/ci-framework-data/logs/crc/crs/glances.yaml

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
recheck

**Comment by fmount** [comment]
recheck

File: config/samples/core_v1beta1_openstackcontrolplane.yaml [hunk 1]
```diff
@@ -57,9 +57,13 @@ spec:
       secret: osp-secret
   glance:
     template:
+      secret: osp-secret
       databaseInstance: openstack
       storageClass: ""
       storageRequest: 10G
+      glanceAPI:
+        type: single
```
Comment (fmount) [inline config/samples/core_v1beta1_openstackcontrolplane.yaml:65]: @abays given we're using statefulset where each instance has stable access to a given pvc, we need to have a "single" instance when "file" is used as a backend

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**